File: schoolDashboard/app/Models/Trip.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Trip extends Model
{
    use HasFactory;
    //allowed columns
    protected $fillable = ['sourceRoute','destinationRoute','startTime','endTime','dateOfTrip','tripStatus'];

    /**
     * on this trip, we assign a student to the van using our created pivot table
     * @return \Illuminate\Database\Eloquent\Relations\BelongsTo<VanStudent, Trip>
     */
    public function vanStudent()
    {
        return $this->belongsTo(VanStudent::class);
    }

    /**
     * assigning the van that is going to make this trip
     */
    public function van()
    {
        return $this->throughVanStudent()->van();
    }

    /**
     * a student is assigned belongs to this trip through the student van pivot table relationship
     */
    public function student()
    {
        return $this->throughVanStudent()->student();
    }

    /**
     * this is the relationship that is acting as the pivot relationship for our pivot table
     */
    protected function throughVanStudent()
    {
        return $this->hasOneDeep(
            [Van::class, Student::class],
            [Trip::class, VanStudent::class],
            [
                'id',          // Trip table
                'id',          // VanStudent table
                'id'          // Van table
            ],
            [
                'van_student_id',  // Trip table
                'van_id',         // VanStudent table
                'id'             // Van table
            ]
        );
    }
}

File: van_operator_app/lib/backend/models/trip.dart
class TripAssigned {
  final int vanId;
  final int vanStudentId;
  final String sourceRoute;
  final String destinationRoute;
  final String startTime;
  final String endTime;
  final String dateOfTrip;
  final String tripStatus;

  TripAssigned({
    required this.vanId,
    required this.vanStudentId,
    required this.sourceRoute,
    required this.destinationRoute,
    required this.startTime,
    required this.endTime,
    required this.dateOfTrip,
    required this.tripStatus,
  });

  factory TripAssigned.fromJson(Map<String, dynamic> json) {
    return TripAssigned(
      vanId: json['vanId'],
      vanStudentId: json['vanStudentId'],
      sourceRoute: json['sourceRoute'],
      destinationRoute: json['destinationRoute'],
      startTime: json['startTime'],
      endTime: json['endTime'],
      dateOfTrip: json['dateOfTrip'],
      tripStatus: json['tripStatus'],
    );
  }
}
